public/editor-client/src/Elements/Filters/index.ts

```diff
@@ -0,0 +1,134 @@
+import { getFields } from "../../api";
+import {
+  Choice,
+  FilterFieldsData,
+  Filters,
+  PossibleValue,
+  QueryTypeSource
+} from "../../types/Filters";
+import { t } from "../../utils/i18n";
+
+export const handler: Filters["handler"] = async (res, rej, data) => {
+  try {
+    const result = await getFields(data);
+
+    const convertedValue = converter(result);
+
+    res(convertedValue ?? []);
+  } catch (e) {
+    rej(t("Failed to load sources"));
+  }
+};
+
+export const possibleValues: Filters["possibleValues"] = async (
+  res,
+  rej,
+  { type, search, optionSource, postId, loopAttributes }
+) => {
+  try {
+    const result = await getFields({ postId, loopAttributes });
+
+    const convertPossibleValues = parseFields(
+      result,
+      optionSource,
+      type,
+      search
+    );
+
+    res(convertPossibleValues ?? []);
+  } catch (e) {
+    rej(t("Failed to load sources"));
+  }
+};
+
+const createQueryTypeSource = (
+  query: string,
+  type: string,
+  source: string,
+  filterBy: string
+): QueryTypeSource => {
+  return `${query}|||${type}|||${source}|||${filterBy}` as QueryTypeSource;
```

Comment:
Make short return and move this function to `utils.ts`.
Also create an reader for `QueryTypeSource` to avoid casting